src/live_restore/live_restore_server.c

```diff
@@ -416,14 +435,15 @@ __wt_live_restore_server_destroy(WT_SESSION_IMPL *session)
     WTI_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
 
     /*
-     * If we didn't create a live restore file system or the server there is nothing to do, it is
-     * rare, but possible, to arrive here with the flag set and a NULL server. This situation
-     * happens when an error is encountered after the file system initialization but before the
-     * server is created.
+     * If we didn't create the background migration server there is nothing to do. It is rare, but
+     * possible, to arrive here with the flag set and a NULL server. This situation arises when an
+     * error is encountered during the server set up.
      */
-    if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS) || server == NULL)
+    if (server == NULL)
         return (0);
 
+    __wt_atomic_storebool(&server->shutting_down, true);
```

Comment:
I would rather do this within the queue_lock that is much more obvious to reason about, you could flag this immediately after a worker thread reaches line live_restore_server.c:118 (the tailq_empty check). With the queue lock we're completely blocked.

Reply:
[e5d8cfa](https://github.com/wiredtiger/wiredtiger/pull/11797/commits/e5d8cfa69b2c037816084abe0bd80853245101ef)